runtime/astar/src/lib.rs

```diff
@@ -663,6 +664,8 @@ parameter_types! {
     pub const DepositPerByte: Balance = contracts_deposit(0, 1);
     // Fallback value if storage deposit limit not set by the user
     pub const DefaultDepositLimit: Balance = contracts_deposit(16, 16 * 1024);
+    pub const MaxDelegateDependencies: u32 = 32;
+    pub const CodeHashLockupDepositPercent: Perbill = Perbill::from_percent(30);
```

Comment:
I understand from the PR summary that this is reused from the original PR.

However, we have a live chain and we should make sure to balance between security & usability.
I'll take a closer look at the PR which introduced the feature, but I hope this won't result in a gas usage bump that users will feel.

Reply:
The `CodeHashLockupDepositPercent` would be charged when code hash ref is increased, i.e two operations.
- **contract add delegate dependency**:  new feature for ensuring that code contract hash is not reaped. Useful for proxies
- **instantiating a contract from code hash**: earlier only base deposit was charged, not the percentage (`CodeHashLockupDepositPercent`) of code deposit is also added to it.


I can do a small test to compare the gas used for
- deploying contract by uploading
- deploying contract by existing code hash